zowex/src/comm.rs

```diff
@@ -155,36 +170,44 @@ pub fn comm_establish_connection(
  *      run by the daemon.
  *      On failure, an error result.
  */
-pub fn comm_talk(message: &[u8], stream: &mut DaemonClient) -> io::Result<i32> {
+pub async fn comm_talk(message: &[u8], stream: &mut DaemonClient) -> io::Result<i32> {
     /*
      * Send the command line arguments to the daemon and await responses.
      */
+    stream.writable().await?;
+
     // write request to daemon
-    stream.write_all(message)?;
+    loop {
+        match stream.try_write(message) {
+            Ok(_n) => break,
+            // If the write operation would need to block, try again as we do not want to discard any messages
+            Err(e) if e.kind() == io::ErrorKind::WouldBlock => continue,
+            Err(e) => {
+                return Err(e.into());
+            }
+        }
+    }
```

Comment:
Instead of the loop, could we use `write_all` here from Tokio io-util?
* https://docs.rs/tokio/latest/tokio/net/struct.UnixStream.html#method.write_all
* https://docs.rs/tokio/latest/tokio/net/windows/named_pipe/struct.NamedPipeClient.html#method.write_all

Reply:
yep! fixed in incoming commit